src/data/equiv/option.lean

```diff
@@ -9,16 +9,50 @@ import control.equiv_functor
 /-!
 # Equivalences for `option α`
 
-We define `remove_none` which acts to provide a `e : α ≃ β` if given a `f : option α ≃ option β`.
 
-To construct an `f : option α ≃ option β` from `e : α ≃ β` such that
-`f none = none` and `f (some x) = some (e x)`, use
-`f = equiv_functor.map_equiv option e`.
+We define
+* `equiv.option_congr`: the `option α ≃ option β` constructed from `e : α ≃ β` by sending `none` to
+  `none`, and applying a `e` elsewhere.
+* `equiv.remove_none`: the `α ≃ β` constructed from `option α ≃ option β` by removing `none` from
+  both sides.
 -/
 
 namespace equiv
 
-variables {α β : Type*} (e : option α ≃ option β)
+variables {α β γ : Type*}
+
+section option_congr
+
+variables
+
+/-- A universe-polymorphic version of `equiv_functor.map_equiv option e`. -/
+@[simps apply]
+def option_congr (e : α ≃ β) : option α ≃ option β :=
+{ to_fun := option.map e,
+  inv_fun := option.map e.symm,
+  left_inv := λ x, (option.map_map _ _ _).trans $
+    e.symm_comp_self.symm ▸ congr_fun option.map_id x,
+  right_inv := λ x, (option.map_map _ _ _).trans $
+    e.self_comp_symm.symm ▸ congr_fun option.map_id x }
+
+@[simp] lemma option_congr_refl : option_congr (equiv.refl α) = equiv.refl _ :=
+ext $ congr_fun option.map_id
+
+@[simp] lemma option_congr_symm (e : α ≃ β) : (option_congr e).symm = option_congr e.symm := rfl
+
+@[simp] lemma option_congr_trans (e₁ : α ≃ β) (e₂ : β ≃ γ) :
+  (option_congr e₁).trans (option_congr e₂) = option_congr (e₁.trans e₂) :=
+ext $ option.map_map _ _
+
+/-- When `α` and `β` are in the same universe, this is the same as the result of
+`equiv_functor.map_equiv`. -/
```

Comment:
so the reason we can't fix `equiv_functor.map_equiv option` is because the `option` instance comes from `is_lawful_traversable`, which (I don't think?) can be sufficiently polymorphic. am I right? whatever the reason is it's worth putting in some docstring somewhere I think

I was going to say lgtm otherwise but you updated a new commit as I did that so i'll check the rest out after I sleep

Reply:
Yeah, there was a zulip thread or two about generalizing all these functor instances. Adding such a docstring seems independent - if we're going to do it, it should be a library_note referenced by all the `control` typeclasses.